src/OrchardCore/OrchardCore.Media.Core/DefaultMediaFileStoreCacheFileProvider.cs

```diff
@@ -67,6 +67,11 @@ public async Task SetCacheAsync(Stream stream, IFileStoreEntry fileStoreEntry, C
                 using (var fileStream = File.Create(cachePath))
                 {
                     await stream.CopyToAsync(fileStream, StreamCopyBufferSize, cancellationToken);
+
+                    if (fileStream.Length == 0)
+                    {
+                        throw new Exception($"Error retrieving file (length equals 0 byte) : {cachePath}");
```

Comment:
This one cool, but also we talked about not giving the stream copy the request cancellationToken. 

So changing to `await stream.CopyToAsync(fileStream, StreamCopyBufferSize, default);` or the appropriate overload.

Let's see the logs you can get from your server with this in it though, so we can see the stack trace.

Reply:
> This one cool, but also we talked about not giving the stream copy the request cancellationToken.

Or, as we did at some point with @Skrypt, check in the following catch `!cancellationToken.IsCancellationRequested` before logging and throwing

                if (!cancellationToken.IsCancellationRequested)
                {
                    _logger.LogError(ex, "Error saving file {Path}", cachePath);
                    throw;
                }

Doing this was reducing logs, but when trying it with @Skrypt we didn't get any 0 length file, so maybe @Skrypt was able to repro and added the length check, but maybe worth to still keep and then check the cancelation token

Reply:
Well we want to keep throwing an exception and there could be an error saving the file without receiving a cancellationToken. We can't rely on the cancellationToken. So, I removed it and yes I need to remove the cancellationToken from the CopyToAsync because here we download the file in the cache. There is no reason to cancel the download of that file.

Reply:
Done I removed the cancellationToken. And now it will only catch the issue where there's a 0 byte fileStream. We just want to track it for now.

Reply:
@Skrypt Okay, no problem, will see

One question, did you be able to repro the 0 byte issue and then reach the line where you check the length?

That would explain that the file would not be deleted (in the catch)

Then about the token it will depend if we want to stop or not the download on a request aborted, if we want to stop the download, good to check again in the catch block if the token has been canceled to not throw.

Reply:
Yeah, we are trying things. I need to move onto something else for the remaining of the week. I might have time to test this back next week. For nice, I keep it on ice. I think we have what we need there. The 0 byte issue is hard to repro ; I need to spam my own website to make it queue up tasks a lot before reaching it.

Reply:
Yep, this bit looks good for trying things, or at least seeing if we can repro it at all.